definitions/features/pulpcore_database.rb

```diff
@@ -7,7 +7,7 @@ class Features::PulpcoreDatabase < ForemanMaintain::Feature
     label :pulpcore_database
 
     confine do
-      file_nonzero?(PULPCORE_DB_CONFIG)
+      file_nonzero?(PULPCORE_DB_CONFIG) && feature(:pulpcore)
```

Comment:
@jlsherrill Similar to pulp2, the pulpcore_database should be available only when pulpcore is available? This is to avoid detection of feature when its not being used, and avoid taking backup of pulpcore_db

Reply:
+1 makes sense

Reply:
As noted yesterday, I think this is not a good idea.

We have scenarios where people will have a Katello running with Pulp2-backed repositories, but Pulp3 is installed (and disabled) for the migration.
I read the code that in this case, the pulpcore db (which is "empty" as in: tables are there, they are fully migrated, but there is no user data) won't be backed up.
If a user will restore such a backup, they end up with no pulpcore DB at all, and then can't do the p2-to-p3 migration due to the missing db (the installer can recreate it just fine, but that's not part of the migration process by default).

Reply:
i thought the restore would run the installer at some point which would deploy an empty (migrated) pulp3 db.

However you bring up a good point, if this is ever run by someone running katello 3.18 i think this logic falls down, as we need to handle both pulp2 and pulp3 dbs

Reply:
@jlsherrill @evgeni handled this situation, request to check